src/proxy/h2/client.rs

```diff
@@ -147,10 +145,10 @@ impl H2ConnectClient {
 
 pub async fn spawn_connection(
     cfg: Arc<config::Config>,
-    s: TlsStream<TcpStream>,
+    s: impl AsyncRead + AsyncWrite + Unpin + Send + 'static,
     driver_drain: Receiver<bool>,
     wl_key: WorkloadKey,
-) -> Result<H2ConnectClient, Error> {
+) -> Result<(H2ConnectClient, tokio::task::JoinHandle<()>), Error> {
```

Comment:
yeah I don't think we need this, doesn't seem like we care too much about gracefule shutdowns